libraries/AP_Soaring/SoaringController.cpp

```diff
@@ -0,0 +1,407 @@
+    // @Range: 0 10
+    // @User: Advanced 
+    AP_GROUPINFO("Q1", 3, SoaringController, thermal_q1, 0.001f),
+        
+    // @Param: Q2
+    // @DisplayName: Process noise
+    // @Description: Standard deviation of noise in process for position and radius
+    // @Units: 
+    // @Range: 0 10
+    // @User: Advanced 
+    AP_GROUPINFO("Q2", 4, SoaringController, thermal_q2, 0.03f),
+    
+    // @Param: R
+    // @DisplayName: Measurement noise
+    // @Description: Standard deviation of noise in measurement
+    // @Units: 
+    // @Range: 0 10
+    // @User: Advanced 
+    
+    AP_GROUPINFO("R", 5, SoaringController, thermal_r, 0.45f),
+    
+    // @Param: DIST_AHEAD
+    // @DisplayName: Distance to thermal center
+    // @Description: Initial guess of the distance to the thermal center
+    // @Units: metres
+    // @Range: 0 100
+    // @User: Advanced 
+    AP_GROUPINFO("DIST_AHEAD", 6, SoaringController, thermal_distance_ahead, 5.0f),
+    
+    // @Param: MIN_THML_S
+    // @DisplayName: Minimum thermalling time
+    // @Description: Minimum number of seconds to spend thermalling
+    // @Units: seconds
+    // @Range: 0 32768
+    // @User: Advanced 
+    AP_GROUPINFO("MIN_THML_S", 7, SoaringController, min_thermal_s, 20),
+    
+    // @Param: MIN_CRSE_S
+    // @DisplayName: Minimum cruising time
+    // @Description: Minimum number of seconds to spend cruising
+    // @Units: seconds
+    // @Range: 0 32768
+    // @User: Advanced 
+    AP_GROUPINFO("MIN_CRSE_S", 8, SoaringController, min_cruise_s, 30),
+    
+    // @Param: POLAR_CD0
+    // @DisplayName: Zero lift drag coef.
+    // @Description: Zero lift drag coefficient
+    // @Units: Non-dim.
+    // @Range: 0 0.5
+    // @User: Advanced 
+    AP_GROUPINFO("POLAR_CD0", 9, SoaringController, polar_CD0, 0.027),
+    
+    // @Param: POLAR_B
+    // @DisplayName: Induced drag coeffient
+    // @Description: Induced drag coeffient
+    // @Units: Non-dim.
+    // @Range: 0 0.5
+    // @User: Advanced 
+    AP_GROUPINFO("POLAR_B", 10, SoaringController, polar_B, 0.031),
+    
+    // @Param: POLAR_K
+    // @DisplayName: Cl factor
+    // @Description: Cl factor 2*m*g/(rho*S)
+    // @Units: m^2 s^-2
+    // @Range: 0 0.5
+    // @User: Advanced 
+    AP_GROUPINFO("POLAR_K", 11, SoaringController, polar_K, 25.6),
+    
+    // @Param: ALT_MAX
+    // @DisplayName: Maximum soaring altitude
+    // @Description: Don't thermal any higher than this.
+    // @Units: m
+    // @Range: 0 1000.0
+    // @User: Advanced 
+    AP_GROUPINFO("ALT_MAX", 12, SoaringController, alt_max, 350.0),
+    
+    // @Param: ALT_MIN
+    // @DisplayName: Minimum soaring altitude
+    // @Description: Don't get any lower than this.
+    // @Units: m
+    // @Range: 0 1000.0
+    // @User: Advanced 
+    AP_GROUPINFO("ALT_MIN", 13, SoaringController, alt_min, 50.0),
+    
+    // @Param: ALT_CUTOFF
+    // @DisplayName: Maximum power altitude
+    // @Description: Cut off throttle at this alt.
```

Comment:
Same comment regarding an uncertainty in what altitude reference frame is used for this.